x-pack/plugins/watcher/server/routes/api/indices/register_get_index_patterns_route.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObject, SavedObjectsFindResult } from 'src/core/server';
+import { RouteDependencies } from '../../../types';
+
+export function registerGetIndexPatternsRoute({
+  router,
+  license,
+  lib: { handleEsError },
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/watcher/indices/index_patterns',
+      validate: false,
+    },
+    license.guardApiRoute(async ({ core: { savedObjects } }, request, response) => {
+      try {
+        const finder = savedObjects.client.createPointInTimeFinder({
+          type: 'index-pattern',
+          fields: ['title'],
+          perPage: 1000,
+        });
+
+        const responses: SavedObject[] = [];
+
+        for await (const result of finder.find()) {
+          responses.push(
+            ...result.saved_objects.map(
+              (indexPattern: SavedObjectsFindResult<any>) => indexPattern.attributes.title
```

Comment:
The client side code only uses the indexPattern names, so we can just return that from the server and avoid sending unnecessary information.